core-primitives/substrate-sgx/environmental/src/lib.rs

```diff
@@ -0,0 +1,479 @@
+// Copyright 2017-2020 Parity Technologies
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Safe global references to stack variables.
+//!
+//! Set up a global reference with environmental! macro giving it a name and type.
+//! Use the `using` function scoped under its name to name a reference and call a function that
+//! takes no parameters yet can access said reference through the similarly placed `with` function.
+//!
+//! # Examples
+//!
+//! ```
+//! #[macro_use] extern crate environmental;
+//! // create a place for the global reference to exist.
+//! environmental!(counter: u32);
+//! fn stuff() {
+//!   // do some stuff, accessing the named reference as desired.
+//!   counter::with(|i| *i += 1);
+//! }
+//! fn main() {
+//!   // declare a stack variable of the same type as our global declaration.
+//!   let mut counter_value = 41u32;
+//!   // call stuff, setting up our `counter` environment as a reference to our counter_value var.
+//!   counter::using(&mut counter_value, stuff);
+//!   println!("The answer is {:?}", counter_value); // will print 42!
+//!   stuff();  // safe! doesn't do anything.
+//! }
+//! ```
+//!
+//! Original crate: https://github.com/paritytech/environmental/blob/master/src/lib.rs
+//! The original crate does not support multithreading in `no_std` mode, see https://github.com/integritee-network/worker/issues/803.
+//! Therefore, this crate introduces the sgx feature, which allows multithreading within an sgx enabled envinronment.
+//! It should be ensured, that all uses of the environmental crate within the enclave are makeing use of this crate, not the original one.
```

Comment:
* Beware language nerd: you almost never have a comma before 'that'.
* typo: `makeing`

Reply:
Always happy to learn from the best 😄